src/main/java/com/ibm/devops/notification/OTCAPIHelper.java

```diff
@@ -0,0 +1,249 @@
+import org.cloudfoundry.client.lib.CloudFoundryClient;
+import org.cloudfoundry.client.lib.HttpProxyConfiguration;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Pattern;
+import java.util.Scanner;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+
+/**
+ * OTCAPIHelper class helps collect various information (org, space, app) from OTC API
+ */
+public abstract class OTCAPIHelper extends Recorder {
+	// API URLs
+    private final static String ORGS_URL= "/v2/organizations?q=name:";
+    private final static String SPACES_URL= "/v2/spaces?q=name:";
+    private final static String APPS_URL= "/v2/apps?q=name:";
+    private final static String ORG= "&&organization_guid:";
+    private final static String SPACE= "&&space_guid:";
+    
+
+    public static void printPluginVersion(ClassLoader loader, PrintStream printStream) {
+        final Properties properties = new Properties();
+        try {
+            properties.load(loader.getResourceAsStream("plugin.properties"));
+            printStream.println("[IBM Cloud DevOps] version: " + properties.getProperty("version"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+    
+    /*
+     * Returns a valid API url (as env variable might be missing the leading https part)
+     */
+    public static String getTargetAPI(EnvVars envVars) {
+    	String cf_api = envVars.get("CF_API");
+    	if (cf_api.startsWith("https")) {
+    		return cf_api;
+    	}
+    	return "https://" + cf_api;
+    }
+    
+    /*
+     * Returns a Bluemix token based on credentials as found in env variables
+     */
+    public static String getBluemixToken(EnvVars envVars, PrintStream printStream) {
+        try {
+        	String targetAPI = getTargetAPI(envVars);
+        	String username = envVars.get("CF_CREDS_USR");
+        	String password = envVars.get("CF_CREDS_PSW");
+        	
+        	CloudCredentials cloudCredentials = new CloudCredentials(username, password);
+
+            URL url = new URL(targetAPI);
+            HttpProxyConfiguration configuration = buildProxyConfiguration(url);
+
+            CloudFoundryClient client = new CloudFoundryClient(cloudCredentials, url, configuration, true);
+            return "bearer " + client.login().toString();
+
+        } catch (Exception e) {
+        	printStream.println("[IBM Cloud DevOps] Unexpected Exception:");
+            e.printStackTrace(printStream);
+        }
+        return "";
+    }
+    
+    /*
+     * Returns all orgs from OTC API for the given Bluemix token
+     */    
+    public static JSONObject getOrgs(EnvVars envVars, String bluemixToken, PrintStream printStream) {
+    	String orgName = envVars.get("CF_ORG");
+    	String orgsUrl = getTargetAPI(envVars) + ORGS_URL + orgName;
```

Comment:
Lately we have been running into issues with special characters in parameters. Please consider URL encoding any parameters that are part of URL.